client/my-sites/plugins/main.jsx

```diff
@@ -207,17 +207,25 @@ export default React.createClass( {
 		);
 	},
 
-	toggleBulkManagement() {
-		const bulkManagement = ! this.state.bulkManagement,
-			sites = bulkManagement ? this.props.sites.getSelectedOrAllWithPlugins() : this.props.sites.getSelectedOrAll();
+	toggleBulkManagement( section ) {
+		let bulkManagement = this.state.bulkManagement;
+		const foundIndex = bulkManagement.indexOf( section );
 
+		if ( foundIndex === -1 ) {
+			bulkManagement.push( section );
+		} else {
+			bulkManagement.splice( foundIndex, 1 );
+		}
+
+		const sites = bulkManagement ? this.props.sites.getSelectedOrAllWithPlugins() : this.props.sites.getSelectedOrAll();
+		console.log( bulkManagement );
 		this.setState( {
 			plugins: this.getPluginsFromStore( this.props, sites ),
 			bulkManagement: bulkManagement,
 			filter: 'all'
 		} );
 
-		if ( bulkManagement ) {
+		if ( bulkManagement.length ) {
```

Comment:
I think we should check for `bulkManagement.indexOf( section ) >= 0` instead of just for length